.github/workflows/contracts.yml

```diff
@@ -0,0 +1,69 @@
+name: Build and test contracts
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./contracts
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Set up cargo cache
+        uses: actions/cache@v3
+        continue-on-error: false
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            lib/target
+          key: 1-${{ runner.os }}-cargo-build-${{ hashFiles('**/Cargo.toml') }}
+          restore-keys: ${{ runner.os }}-cargo-build
+
+      - name: Prepare target
+        run: rustup target add wasm32-unknown-unknown
+
+      - name: Build
+        run: for dir in $(ls); do cd $dir && cargo build --release --target wasm32-unknown-unknown  --verbose || exit 1 ; cd .. ; done
```

Comment:
We are not working with workspaces (we had problems on that previously) , so we need to iterate over each crate.

Reply:
Please just do workspaces here.

Reply:
Initially i was creating it with workspaces, but i was finding problems i would like to investigate in a new issue, separatelly. Unless you have strong opinion though.

Reply:
@geofmureithi , i just added the workspace config at  09f3705e458f899d7e54e48ad4a652e800141a92 , so you can [see the error](https://github.com/eigerco/nebula/actions/runs/5695597389/job/15439060493#step:5:386) i was talking about. Let me know if this sounds to you.